patches/common/chromium/desktop_media_list.patch

```diff
@@ -0,0 +1,287 @@
+ #include "content/public/browser/browser_thread.h"
+ #include "media/base/video_util.h"
+ #include "third_party/libyuv/include/libyuv/scale_argb.h"
+@@ -76,11 +82,13 @@ gfx::ImageSkia ScaleDesktopFrame(std::unique_ptr<webrtc::DesktopFrame> frame,
+ class NativeDesktopMediaList::Worker
+     : public webrtc::DesktopCapturer::Callback {
+  public:
+-  Worker(base::WeakPtr<NativeDesktopMediaList> media_list,
++  Worker(scoped_refptr<base::SingleThreadTaskRunner> task_runner,
++         base::WeakPtr<NativeDesktopMediaList> media_list,
+          DesktopMediaID::Type type,
+          std::unique_ptr<webrtc::DesktopCapturer> capturer);
+   ~Worker() override;
+ 
++  void Start();
+   void Refresh(const DesktopMediaID::Id& view_dialog_id);
+ 
+   void RefreshThumbnails(const std::vector<DesktopMediaID>& native_ids,
+@@ -93,6 +101,9 @@ class NativeDesktopMediaList::Worker
+   void OnCaptureResult(webrtc::DesktopCapturer::Result result,
+                        std::unique_ptr<webrtc::DesktopFrame> frame) override;
+ 
++  // Task runner used for capturing operations.
++  scoped_refptr<base::SingleThreadTaskRunner> task_runner_;
++
+   base::WeakPtr<NativeDesktopMediaList> media_list_;
+ 
+   DesktopMediaID::Type type_;
+@@ -106,17 +117,27 @@ class NativeDesktopMediaList::Worker
+ };
+ 
+ NativeDesktopMediaList::Worker::Worker(
++    scoped_refptr<base::SingleThreadTaskRunner> task_runner,
+     base::WeakPtr<NativeDesktopMediaList> media_list,
+     DesktopMediaID::Type type,
+     std::unique_ptr<webrtc::DesktopCapturer> capturer)
+-    : media_list_(media_list), type_(type), capturer_(std::move(capturer)) {
+-  capturer_->Start(this);
++    : task_runner_(task_runner),
++      media_list_(media_list),
++      type_(type),
++      capturer_(std::move(capturer)) {}
++
++NativeDesktopMediaList::Worker::~Worker() {
++  DCHECK(task_runner_->BelongsToCurrentThread());
+ }
+ 
+-NativeDesktopMediaList::Worker::~Worker() {}
++void NativeDesktopMediaList::Worker::Start() {
++  DCHECK(task_runner_->BelongsToCurrentThread());
++  capturer_->Start(this);
++}
+ 
+ void NativeDesktopMediaList::Worker::Refresh(
+     const DesktopMediaID::Id& view_dialog_id) {
++  DCHECK(task_runner_->BelongsToCurrentThread());
+   std::vector<SourceDescription> result;
+ 
+   webrtc::DesktopCapturer::SourceList sources;
+@@ -133,11 +154,8 @@ void NativeDesktopMediaList::Worker::Refresh(
+         // Just in case 'Screen' is inflected depending on the screen number,
+         // use plural formatter.
+         title = mutiple_sources
+-                    ? l10n_util::GetPluralStringFUTF16(
+-                          IDS_DESKTOP_MEDIA_PICKER_MULTIPLE_SCREEN_NAME,
+-                          static_cast<int>(i + 1))
+-                    : l10n_util::GetStringUTF16(
+-                          IDS_DESKTOP_MEDIA_PICKER_SINGLE_SCREEN_NAME);
++                    ? base::UTF8ToUTF16("Screen " + base::IntToString(i + 1))
++                    : base::UTF8ToUTF16("Entire screen");
```

Comment:
I see this is in the previous version too, but I don't understand it -- why is this patch needed?

Reply:
These `IDS_*` values are generated with appropriate locales during compile time from https://cs.chromium.org/chromium/src/chrome/app/generated_resources.grd. We don't have that generation setup yet, I am actually adding the initial template for a different purpose in https://github.com/electron/electron/pull/14656/commits/22c81fb9d4711549c7bfabf1ae3647a90d448523#diff-50077a5a120716f1f9994e1d326542b2, once thats added, we can add these locale strings and revert this.